prospero-common/src/main/java/org/wildfly/prospero/model/ProsperoConfig.java

```diff
@@ -52,26 +53,52 @@ public List<RepositoryRef> getRepositories() {
     }
 
     public void addChannel(ChannelRef channelRef) {
+        // Check that neither GAV nor URL of added channel is equal to GAVs or URLs of existing channels.
+        if (StringUtils.isNotBlank(channelRef.getGav())) {
+            Optional<ChannelRef> found = channels.stream().filter(c -> channelRef.getGav().equals(c.getGav())).findAny();
+            if (found.isPresent()) {
+                throw Messages.MESSAGES.channelExists(channelRef.getGav());
+            }
+        }
+        if (StringUtils.isNotBlank(channelRef.getUrl())) {
+            Optional<ChannelRef> found = channels.stream().filter(c -> channelRef.getUrl().equals(c.getUrl())).findAny();
+            if (found.isPresent()) {
+                throw Messages.MESSAGES.channelExists(channelRef.getUrl());
+            }
+        }
         channels.add(0, channelRef);
     }
 
+    public void removeChannel(ChannelRef channelRef) {
+        Optional<ChannelRef> found = Optional.empty();
+        if (StringUtils.isNotBlank(channelRef.getGav())) {
+            found = channels.stream().filter(c -> channelRef.getGav().equals(c.getGav())).findFirst();
+        }
+        if (found.isEmpty() && StringUtils.isNotBlank(channelRef.getUrl())) {
+            found = channels.stream().filter(c -> channelRef.getUrl().equals(c.getUrl())).findFirst();
+        }
+        if (found.isEmpty()) {
+            throw Messages.MESSAGES.channelNotPresent(channelRef.getGavOrUrlString());
+        }
+        channels.remove(found.get());
+    }
+
     /**
      * Adds a repository to the config. If the repository already exists ({@code id} and {@code url} matches), it is not added.
      * Adding a repository with existing {@code id} but different {@code url} causes {@code IllegalArgumentException}
      * @param repository
      * @throws IllegalArgumentException if a repository with given id but different URL is already present
      * @return true if the repository was added, false if the repository already exists
      */
-    public boolean addRepository(RepositoryRef repository) {
+    public void addRepository(RepositoryRef repository) {
```

Comment:
I'm not sure about this. Is adding an existing repository a fault? Especially one that results in unchecked exception? A caller might be able to continue - after the call the repositories are in a state they were supposed to be.

I think a conflicting repository (same id, different url) is a different case- something went wrong and we cannot reasonable handle that.

Reply:
OK, that makes sense too :). Dropping the commit.